### PR TITLE
Do not transform jpeg RGB components

### PR DESCRIPTION
**Pull request by havocbcn**
Hello,

Fix issue #11931

The image is encoded using RGB, not YCbCr

here : https://stackoverflow.com/questions/50798014/determining-color-space-for-jpeg/50861048

On ISO/IEC 10918-6:2013 (E), section 6.1: (http://www.itu.int/rec/T-REC-T.872-201206-I/en)

Images encoded with three components are assumed to be RGB data encoded as YCbCr unless the image contains an APP14 marker segment as specified in 6.5.3, in which case the colour encoding is considered either RGB or YCbCr according to the application data of the APP14 marker segment

The head.jpg doesn't have a app14, so image seems to be using YCbCr not RGB

But, the second best answer points to look up the implementation of a Java jpeg library, seems that if components are called R, G and B, the space color is RGB, and this is the case, the components are called in this way.

This pull request maintain the index in the components struct to allow _isColorConversionNeeded to do not convert if the components are called R, G and B.

The Pdf is now showed correctly

### CODE REVIEW

File: src/core/jpg.js [hunk 1]
```diff
@@ -991,8 +991,10 @@ var JpegImage = (function JpegImageClosure() {
             var components = [],
               component;
             for (i = 0; i < selectorsCount; i++) {
-              var componentIndex = frame.componentIds[data[offset++]];
+              var index = data[offset++];
```
Comment (Snuffleupagus) [inline src/core/jpg.js:994]: Nit: Please use `let` or `const`, as appropriate, when adding new code.

**Review by Snuffleupagus — COMMENTED** [review]
Please also update the *commit message* to include all of the relevant information found in https://github.com/mozilla/pdf.js/pull/11967#issue-427623862, such that that information is available on e.g. the Git command-line as well.

Please remember to squash the commits once you've made changes, see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits

File: src/core/jpg.js [hunk 2]
```diff
@@ -1183,6 +1186,14 @@ var JpegImage = (function JpegImageClosure() {
           // dictionary has a 'ColorTransform' entry, explicitly set to `0`,
           // then the colours should *not* be transformed.
           return false;
+        } else if (
+          this.components[0].index === 0x52 && // ASCII R. G and B
+          this.components[1].index === 0x47 &&
+          this.components[2].index === 0x42
```
[the remark this answers is not in the document]
Comment (havocbcn) [reply src/core/jpg.js]: done

File: test/test_manifest.json [hunk 3]
```diff
@@ -4367,6 +4367,12 @@
        "type": "eq",
        "forms": true
     },
+    {  "id": "jpeg_rgb",
+       "file": "pdfs/jpeg_rgb.pdf",
```
[the remark this answers is not in the document]
Comment (havocbcn) [reply test/test_manifest.json]: done

File: test/pdfs/.gitignore [hunk 4]
```diff
@@ -147,6 +147,7 @@
 !issue1055r.pdf
 !issue11713.pdf
 !issue1293r.pdf
+!jpeg_rgb.pdf
```
[the remark this answers is not in the document]
Comment (havocbcn) [reply test/pdfs/.gitignore]: done

File: src/core/jpg.js (hunk 1 above)
Comment (havocbcn) [reply src/core/jpg.js]: done

**Comment by havocbcn** [comment]
Updated commit message, test name, comments and the var to a const.

Thanks

**Review by Snuffleupagus — APPROVED** [review]
Looks good to me, thank you!

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for your contribution!

**Pull request merged**